src/aiida_pseudo/cli/install.py

```diff
@@ -143,117 +140,215 @@ def download_sssp(
+
+    family.description = description
+    family.set_cutoffs(cutoffs, 'normal', unit='Ry')
+
+    echo.echo_success(f'installed `{label}` containing {family.count()} pseudopotentials.')
 
 
 @cmd_install.command('sssp')
 @options.VERSION(type=click.Choice(['1.0', '1.1']), default='1.1', show_default=True)
 @options.FUNCTIONAL(type=click.Choice(['PBE', 'PBEsol']), default='PBE', show_default=True)
 @options.PROTOCOL(type=click.Choice(['efficiency', 'precision']), default='efficiency', show_default=True)
 @options.DOWNLOAD_ONLY()
+@options.FROM_DOWNLOAD()
 @options.TRACEBACK()
 @decorators.with_dbenv()
-def cmd_install_sssp(version, functional, protocol, download_only, traceback):
+def cmd_install_sssp(version, functional, protocol, download_only, from_download, traceback):
     """Install an SSSP configuration.
 
     The SSSP configuration will be automatically downloaded from the Materials Cloud Archive entry to create a new
     `SsspFamily`.
     """
-    # pylint: disable=too-many-locals
+    # pylint: disable=too-many-locals, too-many-statements
     from aiida.common.files import md5_file
-    from aiida.orm import Group, QueryBuilder
 
     from aiida_pseudo import __version__
-    from aiida_pseudo.groups.family import SsspFamily
 
-    from .utils import attempt, create_family_from_archive
+    if download_only and from_download is not None:
+        raise click.BadParameter(
+            'cannot specify both `--download-only` and `--from-download`.',
+            param_hint="'--download-only' / '--from-download'"
+        )
 
     configuration = SsspConfiguration(version, functional, protocol)
-    label = SsspFamily.format_configuration_label(configuration)
 
     if configuration not in SsspFamily.valid_configurations:
-        echo.echo_critical(f'{configuration} is not a valid configuration.')
-
-    if not download_only and QueryBuilder().append(SsspFamily, filters={'label': label}).first():
-        echo.echo_critical(f'{SsspFamily.__name__}<{label}> is already installed')
+        echo.echo_critical(f'{version} {functional} {protocol} is not a valid SSSP configuration')
 
     with tempfile.TemporaryDirectory() as dirpath:
 
         dirpath = pathlib.Path(dirpath)
+
         filepath_archive = dirpath / 'archive.tar.gz'
         filepath_metadata = dirpath / 'metadata.json'
 
-        patch_version = download_sssp(configuration, filepath_archive, filepath_metadata, traceback)
+        if from_download is not None:
 
-        description = f'SSSP v{patch_version} {functional} {protocol} installed with aiida-pseudo v{__version__}'
-        description += f'\nArchive pseudos md5: {md5_file(filepath_archive)}'
-        description += f'\nPseudo metadata md5: {md5_file(filepath_metadata)}'
+            tarball_path = pathlib.Path(from_download).absolute()
+            with tarfile.open(tarball_path, 'r') as handle:
+                handle.extractall(dirpath)
```

Comment:
It is not immediately clear from reading the code that this assumes that the extracted archive will have written the `filepath_archive` and `filepath_metadata` files. Maybe it would help to have an assert, or just a check `if not filepath_archive.exists(): echo.echo_critical('archive `{from_download}` did not contain required file `{filepath_archive.name}`` and same for `filepath_metadata`